HadesScheduler/HadesOperator/internal/controller/buildjob_controller.go

```diff
@@ -323,3 +396,117 @@ func jobFinished(k8sJob *batchv1.Job) (done bool, succeeded bool, reason string)
 	}
 	return false, false, ""
 }
+
+func jobDone(k8sJob *batchv1.Job) (done bool, succeeded bool, reason string) {
+	for _, c := range k8sJob.Status.Conditions {
+		switch c.Type {
+		case batchv1.JobComplete:
+			if c.Status == corev1.ConditionTrue {
+				return true, true, c.Message
+			}
+		case batchv1.JobFailed:
+			if c.Status == corev1.ConditionTrue {
+				return true, false, c.Message
+			}
+		}
+	}
+	return false, false, ""
+}
+
+// countActiveJobs counts the number of non-completed, non-suspended Jobs in the given namespace.
+func (r *BuildJobReconciler) countActiveJobs(ctx context.Context, namespace string) (int, error) {
+	var jl batchv1.JobList
+	if err := r.List(ctx, &jl,
+		client.InNamespace(namespace),
+		client.MatchingLabels{"hades.tum.de/managed-by": "hades-operator"},
+	); err != nil {
+		return 0, err
+	}
+	active := 0
+	for _, j := range jl.Items {
+		done, _, _ := jobDone(&j)
+		if done {
+			continue
+		}
+		// Unless the job is completed or suspended, count it as active
+		if j.Spec.Suspend == nil || !*j.Spec.Suspend {
+			active++
+		}
+	}
+	return active, nil
+}
+
+// admitOneSuspendedJob unsuspends the oldest suspended Job in the given namespace, if any.
+func (r *BuildJobReconciler) admitOneSuspendedJob(ctx context.Context, namespace string) error {
+	var jl batchv1.JobList
+	if err := r.List(ctx, &jl,
+		client.InNamespace(namespace),
+		client.MatchingLabels{LabelManagedBy: "hades-operator"},
+	); err != nil {
+		return err
+	}
+
+	var pick *batchv1.Job
+	bestPri := -1
+
+	for i := range jl.Items {
+		j := &jl.Items[i]
+
+		done, _, _ := jobDone(j)
+		if done {
+			continue
+		}
+		if j.Spec.Suspend == nil || !*j.Spec.Suspend {
+			continue
+		}
```

Comment:
Can that be not formed into one single statement? 

```suggestion
		if done || j.Spec.Suspend == nil || !*j.Spec.Suspend {
			continue
		}
```